File: Backend/msProduct/internal/domain/models/product.go
package models

import "go.mongodb.org/mongo-driver/bson/primitive"

type Product struct {
	ID            primitive.ObjectID `bson:"_id,omitempty" json:"id,omitempty"`
	Name          string             `bson:"name" json:"name"`
	Description   string             `bson:"description" json:"description"`
	Price         float64            `bson:"price" json:"price"`
	ImageURLs     []string           `bson:"imageURLs" json:"imageURLs"` 
}

File: Backend/ms-cart/src/main/java/com/example/msCart/internal/domain/models/Product.java
package com.example.msCart.internal.domain.models;

import jakarta.persistence.Transient;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.List;

@AllArgsConstructor
@NoArgsConstructor
@Setter
@Getter
public class Product {

        private String id;
        private String name;
        private Integer price;
        private String description;
        private List<String> ImageURLs;
        private Integer amount;
        private String subCategoryId;
        @Transient
        private Integer quantityAvailable;

}
